lib/rpcserver/RpcMethods.js

```diff
@@ -36,6 +47,27 @@ class RpcMethods {
   async connect(params) {
     await this.p2p.connect(params.host, params.port);
   }
+
+  async shutdown() {
+    // ensure we stop listening for new peers before disconnecting from peers
+    if (this.p2pServer) {
+      await this.p2pServer.close();
+    }
+    this.p2p.endAll();
+
+    // TODO: ensure we are not in the middle of executing any trades
+
+    const msg = 'XUD shutdown gracefully';
+    (async () => {
+      // we use an immediately invoked function here to close rpcServer and exit process AFTER the
+      // shutdown method returns a response.
+      await this.rpcServer.close();
+      this.logger.info(msg);
+      process.exit();
```

Comment:
@moshababo Yes I'm going for shutting down all of `xud`. I left out the DB because the documentation said that connections are closed automatically upon the process exiting, but I can put it in there explicitly if you'd rather.

I'd want the daemon to shutdown after this call, I'll look into a cleaner way of doing it than `process.exit()` but let me know if you have any ideas.

Reply:
@moshababo One idea is to move the `shutdown` method to the xuc class, pass that function to the `rpcServer` instance, and then call that. Probably makes the most sense since the "shutdown" is supposed to be for the full node, and it also means that if we ever run an rpc server standalone process we can the shutdown method only shut down the rpc server. Seems like the best way to go to me but just wanted to run it by you.

Reply:
See latest changes.